dll/win32/shdocvw/CNSCBand.cpp

```diff
@@ -0,0 +1,1724 @@
+{
+    // We do nothing, we don't have anything to save yet
+    TRACE("CloseDW called\n");
+    return S_OK;
+}
+
+STDMETHODIMP CNSCBand::ResizeBorderDW(const RECT *prcBorder, IUnknown *punkToolbarSite, BOOL fReserved)
+{
+    /* Must return E_NOTIMPL according to MSDN */
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CNSCBand::ShowDW(BOOL fShow)
+{
+    m_fVisible = fShow;
+    ShowWindow(fShow ? SW_SHOW : SW_HIDE);
+    return S_OK;
+}
+
+// *** IDeskBand ***
+
+STDMETHODIMP CNSCBand::GetBandInfo(DWORD dwBandID, DWORD dwViewMode, DESKBANDINFO *pdbi)
+{
+    if (!pdbi)
+        return E_INVALIDARG;
+
+    m_dwBandID = dwBandID;
+
+    if (pdbi->dwMask & DBIM_MINSIZE)
+    {
+        pdbi->ptMinSize.x = 200;
+        pdbi->ptMinSize.y = 30;
+    }
+
+    if (pdbi->dwMask & DBIM_MAXSIZE)
+        pdbi->ptMaxSize.y = -1;
+
+    if (pdbi->dwMask & DBIM_INTEGRAL)
+        pdbi->ptIntegral.y = 1;
+
+    if (pdbi->dwMask & DBIM_ACTUAL)
+    {
+        pdbi->ptActual.x = 200;
+        pdbi->ptActual.y = 30;
+    }
+
+    if (pdbi->dwMask & DBIM_TITLE)
+    {
+        _GetTitle(pdbi->wszTitle, _countof(pdbi->wszTitle));
+    }
+
+    if (pdbi->dwMask & DBIM_MODEFLAGS)
+        pdbi->dwModeFlags = DBIMF_NORMAL | DBIMF_VARIABLEHEIGHT;
+
+    if (pdbi->dwMask & DBIM_BKCOLOR)
+        pdbi->dwMask &= ~DBIM_BKCOLOR;
+
+    return S_OK;
+}
+
+// *** IObjectWithSite ***
+
+STDMETHODIMP CNSCBand::SetSite(IUnknown *pUnkSite)
+{
+    HRESULT hr;
+
+    if (pUnkSite == m_pSite)
+        return S_OK;
+
+    TRACE("SetSite called\n");
+
+    if (!pUnkSite)
+    {
+        DestroyWindow();
+        m_hWnd = NULL;
+    }
+
+    if (pUnkSite != m_pSite)
+        m_pSite = NULL;
+
+    if (!pUnkSite)
+        return S_OK;
+
+    HWND hwndParent;
+    hr = IUnknown_GetWindow(pUnkSite, &hwndParent);
+    if (!SUCCEEDED(hr))
+    {
+        ERR("Could not get parent's window! 0x%08lX\n", hr);
+        return E_INVALIDARG;
+    }
+
+    m_pSite = pUnkSite;
+
+    if (m_hWnd)
+    {
+        SetParent(hwndParent); // Change its parent
+    }
+    else
+    {
+        this->Create(hwndParent, NULL, NULL, WS_CHILD | WS_VISIBLE, 0, (UINT)0, NULL);
+    }
+
+    _Init();
+
+    return S_OK;
+}
+
+STDMETHODIMP CNSCBand::GetSite(REFIID riid, void **ppvSite)
+{
+    if (!ppvSite)
+        return E_POINTER;
+    *ppvSite = m_pSite;
+    return S_OK;
+}
+
+// *** IOleCommandTarget ***
+
+STDMETHODIMP CNSCBand::QueryStatus(const GUID *pguidCmdGroup, ULONG cCmds, OLECMD prgCmds [], OLECMDTEXT *pCmdText)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+STDMETHODIMP CNSCBand::Exec(const GUID *pguidCmdGroup, DWORD nCmdID, DWORD nCmdexecopt, VARIANT *pvaIn, VARIANT *pvaOut)
+{
+    UNIMPLEMENTED;
+    return E_NOTIMPL;
+}
+
+// *** IServiceProvider ***
+
+STDMETHODIMP CNSCBand::QueryService(REFGUID guidService, REFIID riid, void **ppvObject)
+{
+    return IUnknown_QueryService(m_pSite, guidService, riid, ppvObject);
+}
+
+// *** IServiceProvider ***
```

Comment:
IContextMenu? And if all methods return E_NOTIMP the entire thing can be removed...

Reply:
Comment fixed. The `IContextMenu` interface is exposed in both `CExplorerBand` and `CFavBand`, so I think it has a special meaning.

Reply:
In `CFavBand`, there are the `"Expand"` and `"Create new folder"` menu items.